feg/gateway/services/envoy_controller/control_plane/control_plane.go

```diff
@@ -0,0 +1,364 @@
+
+	srv := xds.NewServer(ctx, cli.config, cb)
+
+	// start the xDS server
+	go RunManagementServer(ctx, srv, port)
+	go RunManagementGateway(ctx, srv, gatewayPort)
+
+	<-signal
+
+	cb.Report()
+
+	return &cli
+}
+
+func getHttpConnectionManager(routeConfigName string, virtualHosts []*v2route.VirtualHost) *hcm.HttpConnectionManager {
+	return &hcm.HttpConnectionManager{
+		CodecType:        hcm.HttpConnectionManager_AUTO,
+		StatPrefix:       "ingress_http",
+		UseRemoteAddress: &wrappers.BoolValue{Value: true},
+		CommonHttpProtocolOptions: &core.HttpProtocolOptions{
+			IdleTimeout: ptypes.DurationProto(idleTimeout),
+			// TODO figure out why this doesn't work properly
+			//HeadersWithUnderscoresAction: core.HttpProtocolOptions_REJECT_REQUEST,
+		},
+		Http2ProtocolOptions: &core.Http2ProtocolOptions{
+			MaxConcurrentStreams:        &wrappers.UInt32Value{Value: maxConcurrentStreams},
+			InitialStreamWindowSize:     &wrappers.UInt32Value{Value: initialStreamWindowSize},
+			InitialConnectionWindowSize: &wrappers.UInt32Value{Value: initialConnectionWindowSize},
+		},
+		RouteSpecifier: &hcm.HttpConnectionManager_RouteConfig{
+			RouteConfig: &v2.RouteConfiguration{
+				Name:         routeConfigName,
+				VirtualHosts: virtualHosts,
+			},
+		},
+		HttpFilters: []*hcm.HttpFilter{{
+			Name: wellknown.Router,
+		}},
+	}
+}
+
+func getVirtualHost(virtualHostName string, domains []string, requestHeadersToAdd []*core.HeaderValueOption) *v2route.VirtualHost {
+	return &v2route.VirtualHost{
+		Name:                virtualHostName,
+		Domains:             domains,
+		RequestHeadersToAdd: requestHeadersToAdd,
+		Routes: []*v2route.Route{{
+			Match: &v2route.RouteMatch{
+				PathSpecifier: &v2route.RouteMatch_Prefix{
+					Prefix: targetPrefix,
+				},
+			},
+			Action: &v2route.Route_Route{
+				Route: &v2route.RouteAction{
+					ClusterSpecifier: &v2route.RouteAction_Cluster{
+						Cluster: clusterName,
+					},
+				},
+			},
+		}}}
+}
+
+func (cli *ControllerClient) UpdateSnapshot(ues []*protos.AddUEHeaderEnrichmentRequest) {
+	nodeId := cli.config.GetStatusKeys()[0]
+
+	var listenerName = "default_http"
+	var virtualHostName = "local_service"
+	var routeConfigName = "matched_website_route"
+	var defaultRouteName = "default_route"
+
+	cluster := []cache.Resource{
+		&v2.Cluster{
+			Name:                 clusterName,
+			ClusterDiscoveryType: &v2.Cluster_Type{Type: v2.Cluster_ORIGINAL_DST},
+			ConnectTimeout:       ptypes.DurationProto(connectTimeout),
+			LbPolicy:             v2.Cluster_CLUSTER_PROVIDED,
+		},
+	}
+	filterChains := []*listener.FilterChain{}
+	virtualHosts := []*v2route.VirtualHost{getVirtualHost(virtualHostName, []string{"*"}, []*core.HeaderValueOption{})}
+
+	pbst, err := ptypes.MarshalAny(getHttpConnectionManager(defaultRouteName, virtualHosts))
+	if err != nil {
+		glog.Errorf("Couldn't marshal HTTP connection manager")
+		return
+	}
+	filterChains = append(filterChains, &listener.FilterChain{
```

Comment:
I would define `filterChains` before this line. 
As a general rule, try to define the vars close where you are going to use them